tests/integration/sparse_pipeline/test_sparse_pipeline.py

```diff
@@ -29,37 +26,33 @@ def docs_to_index(num_docs):
     return DocumentArray(docs)
 
 
-class DummySparseEncoder(Executor):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-
-    def encode(self, content: 'np.ndarray', *args, **kwargs) -> Any:
-        embed = sparse.csr_matrix(content)
-        return embed
-
-
-class DummyCSRSparseIndexer(Executor):
+class DummyCSRSparseIndexEncoder(Executor):
     embedding_cls_type = 'scipy_csr'
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.keys = []
+        self.docs = []
         self.vectors = {}
 
-    def add(
-        self, keys: Iterable[str], vectors: 'scipy.sparse.csr_matrix', *args, **kwargs
-    ) -> None:
-        assert isinstance(vectors, sparse.csr_matrix)
-        self.keys.extend(keys)
-        for i, key in enumerate(keys):
-            self.vectors[key] = vectors.getrow(i)
+    @requests(on='index')
+    def encode(self, docs: 'DocumentArray', *args, **kwargs) -> Any:
```

Comment:
this will not work to have 2 methods with the same request. Maybe have a single privat encode method that u call from add and search